File: common/src/main/java/org/sandboxpowered/silica/network/login/serverbound/LoginStart.java
package org.sandboxpowered.silica.network.login.serverbound;

import org.sandboxpowered.silica.network.Connection;
import org.sandboxpowered.silica.network.Packet;
import org.sandboxpowered.silica.network.PacketByteBuf;
import org.sandboxpowered.silica.network.PacketHandler;

public class LoginStart implements Packet {
    private String username;

    @Override
    public void read(PacketByteBuf buf) {
        username = buf.readString(16);
    }

    @Override
    public void write(PacketByteBuf buf) {
        buf.writeString(username);
    }

    @Override
    public void handle(PacketHandler packetHandler, Connection connection) {
        connection.handleLoginStart(username);
    }
}

File: common/src/main/kotlin/org/sandboxpowered/silica/network/login/serverbound/LoginStart.kt
package org.sandboxpowered.silica.network.login.serverbound

import org.sandboxpowered.silica.network.Connection
import org.sandboxpowered.silica.network.Packet
import org.sandboxpowered.silica.network.PacketByteBuf
import org.sandboxpowered.silica.network.PacketHandler

class LoginStart(private var username: String?) : Packet {

    constructor() : this(null)

    override fun read(buf: PacketByteBuf) {
        username = buf.readString(16)
    }

    override fun write(buf: PacketByteBuf) {
        buf.writeString(username!!)
    }

    override fun handle(packetHandler: PacketHandler, connection: Connection) {
        connection.handleLoginStart(username!!)
    }
}